src/commands/image/imageSource/ContainerAppUpdateStep.ts

```diff
@@ -51,13 +53,35 @@ export class ContainerAppUpdateStep<T extends ImageSourceContext & IngressContex
             name: getContainerNameForImage(nonNullProp(context, 'image')),
         });
 
-        const updating = localize('updatingContainerApp', 'Updating container app...', containerApp.name);
-        progress.report({ message: updating });
+        const retries = 4;
+        await retry(
+            async (currentAttempt: number): Promise<void> => {
+                if (context.newRegistryCredentialType === RegistryCredentialType.DockerLogin && currentAttempt === 2) {
+                    const reason: string = localize('authenticationRequired', 'Container registry authentication was rejected due to unauthorized access. This may be due to internal permissions still propagating. Authentication will be attempted up to {0} times.', retries + 1);
+                    ext.outputChannel.appendLog(reason);
+                }
 
-        await ext.state.runWithTemporaryDescription(containerApp.id, localize('updating', 'Updating...'), async () => {
-            context.containerApp = await updateContainerApp(context, context.subscription, containerAppEnvelope);
-            ext.state.notifyChildrenChanged(containerApp.managedEnvironmentId);
-        });
+                const message: string = currentAttempt === 1 ?
+                    localize('updatingContainerApp', 'Updating container app...') :
+                    localize('updatingContainerAppAttempt', 'Updating container app (Attempt {0}/{1})...', currentAttempt, retries + 1);
+                progress.report({ message: message });
+                ext.outputChannel.appendLog(message);
+
+                await ext.state.runWithTemporaryDescription(containerApp.id, localize('updating', 'Updating...'), async () => {
+                    context.containerApp = await updateContainerApp(context, context.subscription, containerAppEnvelope);
+                    ext.state.notifyChildrenChanged(containerApp.managedEnvironmentId);
+                });
+            },
+            {
+                onFailedAttempt: (err: retry.FailedAttemptError) => {
+                    if (!/authentication\srequired/i.test(err.message)) {
```

Comment:
Not sure if I should only run retries after seeing "authentication required" in the error message, or if I should also check that admin credential-based auth was selected.